### PR TITLE
[BOP-313] Add documentation for manifest kustomize support

### PR DESCRIPTION
**Pull request by sakshisharma84**
This PR adds documentation for kustomize support introduced for manifest based addons.
JIRA Ticket: https://mirantis.jira.com/browse/BOP-313

### CODE REVIEW

**Comment by sakshisharma84** [comment]
@nwneisen Could you please give it a try and let me know if you find any gaps?

**Review by nwneisen — APPROVED** [review]
LGTM

**Review by ranyodh — APPROVED** [review]

File: website/content/docs/blueprint-reference/components.md [hunk 1]
```diff
@@ -128,4 +130,99 @@ spec:
...
+
+Field                   | Description                                                                                                           |
+|:------------------------|:----------------------------------------------------------------------------------------------------------------------|
+| manifest.values.patches | Used to specify list of inline patches to add or override manifest objects(Optional) [Example Usage](#inline-patches) |
+| manifest.values.images  | Used to specify list of images to be replaced in the manifest(Optional)  [Example Usage](#images)                     |
+
+
+### Inline Patches
+
+The users can specify one or more inline patches to add or override manifest objects.
```
Comment (ranyodh) [inline website/content/docs/blueprint-reference/components.md:149]: **suggestion**: To provide a bit more context on what Kustomize options are supported, may be we can add the following here:

---
The users can now customize the static resources for URL based Manifest addons. These customization are based on [Kustomize](https://kustomize.io/).

Following `kustomize` primitives are currently supported:
* [Built-in images transformer](https://github.com/MirantisContainers/boundless-cli/blob/v0.2.0/images).
* [Inline patches](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/)

File: website/content/docs/blueprint-reference/components.md [hunk 2]
```diff
@@ -128,4 +130,99 @@ spec:
         enabled: true
         manifest:
           url: "https://raw.githubusercontent.com/metallb/metallb/v0.14.3/config/manifests/metallb-native.yaml"
-```
\ No newline at end of file
+```
+
+### Kustomize Manifest based addons
+
+The users can now apply customizations on top of the url based manifests. They can specify `inline patches` and keyword-based `images` kustomization in the blueprint. The boundless controller applies these kustomize transformations on the manifest objects and deploys them in the cluster.
```
Comment (ranyodh) [inline website/content/docs/blueprint-reference/components.md]: **suggestion**: To provide a bit more context on what Kustomize options are supported, may be we can add the following here:

---
The users can now customize the static resources for URL based Manifest addons. These customization are based on [Kustomize](https://kustomize.io/).

Following `kustomize` primitives are currently supported:
* [Built-in images transformer](https://github.com/MirantisContainers/boundless-cli/blob/v0.2.0/images).
* [Inline patches](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/)

**Pull request merged**